server/channels/app/scheduled_post.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+)
+
+func (a *App) SaveScheduledPost(rctx request.CTX, scheduledPost *model.ScheduledPost) (*model.ScheduledPost, *model.AppError) {
+	scheduledPost.Id = model.NewId()
+
+	if scheduledPost.ScheduledAt < model.GetMillis() {
```

Comment:
I didn't see this with my other comment, but does that means we're validating twice?

Reply:
Ah yes, this shouldn't be here. Fixed.